thunder/executors/nvfuserex_impl.py

```diff
@@ -412,6 +482,14 @@ def compute_contiguity(
     return tuple(tuple(x) for x in nv_compute_td(shape, stride))
 
 
+def make_key_from_dtensor(dtensor: DTensor) -> tuple[str]:
+    if isinstance(dtensor, DTensor):
+        key = (repr(dtensor.device_mesh), repr(dtensor.placements))
```

Comment:
Both DeviceMesh and tuple[Placement, ...] are hashable and comparable. There's no need to use their string representation for comparison.
```suggestion
        key = (dtensor.device_mesh, dtensor.placements)
```